Allow to configure aria-live message

gribnoysup opened the pull request:
### Description

This change adds new `renderAnnounceSlideMessage` prop to the `Carousel` component. It allows customizing announcement message for the localization purposes. There is also a default fallback that just keeps the current behaviour.

I'm not sure if this is the best API for this kind of functionality (this is why I didn't update the docs yet) and happy to change it based on maintainers suggestion, but I feel like this is a needed change for users of this library in non-English speaking countries 😊

#### Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How Has This Been Tested?

- Added unit test to test the default behaviour
- Manually checked in the browser with the demo

I didn't find any e2e tests for this part of the carousel, but I'm happy to add them if you feel that this is needed 

### Checklist: (Feel free to delete this section upon completion)

- [x] My code follows the style guidelines of this project (I have run `yarn lint`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (I have run `yarn test` and `yarn test-e2e`)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

sarmeyer commented:
@gribnoysup This is great! Thank you for submitting this PR. Would you mind adding this new prop to the Readme prop list? This way, we can get all the changes merged in one PR

gribnoysup commented:
@sarmeyer Thanks for the feedback! I added the new prop to the props table and also added a small explanation section. I had to run the table through Prettier and the raw diff looks a lot scarier than it should, sorry for that 🙏The `rich diff` thingie in GitHub changes preview better shows the number of real changes 😅

sarmeyer commented:
Looks great - thanks again!
